packages/market-service/src/coingecko/coingecko.ts

```diff
@@ -100,11 +105,19 @@ export class CoinGeckoMarketService implements MarketService {
 
       const currency = 'usd'
       const marketData = data?.market_data
+      /* max_supply may be null on coingecko while available on other sources (coincap)
+      hence choosing to take value from total_supply if existing
+      Also a lot of time when max_supply is null, total_supply is the maximum supply on coingecko
+      We can reassess in the future the degree of precision we want on that field */
```

Comment:
Given this issue, we may consider moving coincap to the beginning of the list in data sources to read from.

Before, we never really had any reason that coingecko was the first data source other than it was the first one implemented.

If you are finding in your testing that coincap is more reliable in returning some of this data and if moving to the first data source doesn't cause any obvious issues, this would be a good reason to do that.

Would just be moving coincap above coingecko in `packages/market-service/src/market-providers.ts`

Reply:
@theoboldfrazier That is a good suggestion and I actually contemplated doing it while developping this. Can I create a separate issue for this ? I am afraid doing it now would increase the scope of our issue maybe a bit too much.